core/src/main/java/com/linecorp/armeria/client/limit/ConcurrencyLimit.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.limit;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Predicate;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+
+/**
+ * Concurrency settings that limits the concurrent number of active requests.
+ */
+public final class ConcurrencyLimit {
+    /**
+     * Returns a new builder.
+     */
+    public static ConcurrencyLimitBuilder builder() {
+        return new ConcurrencyLimitBuilder();
+    }
+
+    private final Predicate<ClientRequestContext> policy;
+    private final int maxConcurrency;
+    private final long timeoutMillis;
+    private final AtomicInteger numActiveRequests = new AtomicInteger();
+
+    ConcurrencyLimit(Predicate<ClientRequestContext> policy, int maxConcurrency, long timeoutMillis) {
+        this.policy = policy;
+        this.maxConcurrency = maxConcurrency;
+        this.timeoutMillis = timeoutMillis;
+    }
+
+    /**
+     * Returns the value of the {@code "maxConcurrency"}.
+     */
+    public int maxConcurrency() {
+        return maxConcurrency;
+    }
+
+    /**
+     * Returns the value of the {@code "timeout"} with the desired {@code "unit"}.
+     */
+    public long timeout(TimeUnit desiredUnit) {
```

Comment:
How about removing this method because it's not used?
We can add this later if it's needed. 😄